src/dev/dev-server.ts

```diff
@@ -107,6 +121,9 @@ export const startDevServer = async (options: StartDevServerOptions) => {
                 )
               }
 
+              isBuilding = false
+              buildEvents.emit("built")
+
               options.onBuildEnd?.()
```

Comment:
could be simplified by emitting `built` and `building`, and setting the variables/calling `options` as an event handler

Reply:
if we publicly expose `buildEvents` that would make sense, but going to stick with the current implementation for now to slightly reduce depth